__TESTS__/unit/config/cloudinaryConfig.test.ts

```diff
@@ -152,4 +155,167 @@ describe('Tests for CloudinaryConfiguration', () => {
 
     expect(url).toContain('http://');
   });
-});
+  it('should allow overriding cloudName in config', () => {
+    const conf = new CloudinaryConfig({
+      cloud: {
+        cloudName: 'foo'
+      }
+    });
+
+    conf.cloud.cloudName = 'demo';
+
+    expect(conf.cloud).toEqual({
+      cloudName: 'demo'
+    });
+  });
+
+  // These tests should be "translated" to js base code when these missing features are added:
+  it.skip('should use resource_type from options', () => {
+    test_cloudinary_url('test', {
+      resource_type: 'raw'
+    }, 'https://res.cloudinary.com/test123/raw/upload/test', {});
+  });
+
+  it.skip('should support external cname with cdn_subdomain on', () => {
+    test_cloudinary_url('test', {
+      cname: 'hello.com',
```

Comment:
you can implement this test using the urlConfig.
See the URL config class:
https://github.com/cloudinary/cloudinary-js-base/blob/master/src/config/URLConfig.ts

And how it's used to create the URL
https://github.com/cloudinary/cloudinary-js-base/blob/master/src/internal/url/cloudinaryURL.ts